vmdb/app/models/user.rb

```diff
@@ -915,6 +915,11 @@ def self.current_userid=(userid)
     Thread.current[:userid] = userid
   end
 
+  def self.current_user=(user)
+    Thread.current[:user]   = user
+    Thread.current[:userid] = user.try(:userid)
+  end
```

Comment:
Actually there are no callers to this in your code changes, so why add it?  (I do understand that's TODO, but then this code should be TODO as well).